hypothesis/h
# Support for metadata in new annotations

Support for metadata in a JWE in new annotations. 

After this we'd need to:

- Support editing: https://github.com/hypothesis/h/pull/8182
- Send the metadata from LMS with the right data
- Expose the metadata in the bulk annotation API

That should be enough to enable the assignment breakdown in the email digest.


## Testing 

- Check the API works as before with no metadata

Make an annotation in https://hypothesis.instructure.com/courses/125/assignments/873


- Copy the curl request from the LMS client call to H, the POST to `http://localhost:5000/api/annotations`

edit the --data-raw parameter to include `metadata` with the JWE generated above, eg:

```"metadata": {"assignment_id": 1}```


- Check the metadata is now in the DB:

```
 annotation_id |             data              
---------------+-------------------------------
            10 | {"lms": {"assignment_id": 1}}
(1 row)
```
